src/Framework/LazyFormattedBuildEventArgs.cs

```diff
@@ -17,33 +16,34 @@ public class LazyFormattedBuildEventArgs : BuildEventArgs
         /// <summary>
         /// Stores the message arguments.
         /// </summary>
-        private object[] arguments;
+        private volatile object argumentsOrFormattedMessage;
```

Comment:
Out of curiosity, why did you decide to store the union for both here, instead of utilizing RawMessage to access the message field in the base type?

Reply:
For thread-safety I need the formatting to result in only one reference write (guaranteed to be atomic by the platform). Swapping one string for another wouldn't work because you wouldn't know if the string you're looking at has already been formatted or not. So I swap the arguments array for the new string, for which I need to emulate a union. It is possible that I am missing something, please elaborate if I sound off base :smile: